Makefile.am

```diff
@@ -0,0 +1,32 @@
+AUTOMAKE_OPTIONS = subdir-objects
```

Comment:
Personally I prefer recursive subdirs, but it's only a matter of preference

Reply:
recursive subdirs? Could you please expand on that?

Regarding `subdir-objects`, I'm not sure it's necessary and beneficial. Doing builds outside of the source tree with the products all ending up in the build tree is nice and `make distcheck` nicely checks that it works properly (i.e. that `$(srcdir)` and `$(builddir)` are used properly.

Reply:
> recursive subdirs? Could you please expand on that?

Oh, you mean `SUBDIRS` is missing here?

Reply:
I leave this for you to rework @vpodzime :+1: